src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -108,6 +108,41 @@ record = Marshal(item);
             }
         }
 
+        public override async Task<DValue<BooleanValue>> RemoveAsync(IEnumerable<FormulaValue> recordsToRemove, bool all = false)
+        {
+            var ret = false;
+
+            if (_sourceList == null || _sourceList.IsReadOnly)
```

Comment:
>_sourceList.IsReadOnly) [](http://example.com/codeflow?start=39&length=23)

nit: we should check this in the ctor
if (_sourceList.IsReadOnly) { _sourceList  = null}

that can simplify the check on line 78 as well.